web/src/components/pages/dashboard/stats/stats-card.tsx

```diff
@@ -26,11 +31,16 @@ export default function StatsCard({
   scrollable?: boolean;
   currentLocale?: string;
 }) {
-  const [locale, setLocale] = useState(ALL_LOCALES);
+  const [locale, setLocale] = useLocalStorageState(DEFAULT_LOCALE_OPTION, id);
   const [selectedTab, setSelectedTab] = useState(Object.keys(tabs)[0]);
-  useEffect(() => setLocale(currentLocale ? currentLocale : ALL_LOCALES), [
-    currentLocale,
-  ]);
+  const isDefaultOptionSelected = locale === DEFAULT_LOCALE_OPTION;
+
+  // handle site wide locale change
+  useEffect(() => {
+    if (isDefaultOptionSelected && currentLocale) {
```

Comment:
Without this check, locale changes overwrite our localstorage change as it would expect a page refresh to revert to the default option.